pkg/execute/source.go

```diff
@@ -0,0 +1,86 @@
+package execute
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sort"
+	"text/tabwriter"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/kubeshop/botkube/pkg/bot/interactive"
+	"github.com/kubeshop/botkube/pkg/config"
+	"github.com/kubeshop/botkube/pkg/execute/command"
+)
+
+var (
+	sourceFeatureName = FeatureName{
+		Name:    "source",
+		Aliases: []string{"sources", "src"},
+	}
+)
+
+// SourceExecutor executes all commands that are related to sources.
+type SourceExecutor struct {
+	log               logrus.FieldLogger
+	analyticsReporter AnalyticsReporter
+	cfg               config.Config
+}
+
+// NewSourceExecutor returns a new SourceExecutor instance.
+func NewSourceExecutor(log logrus.FieldLogger, analyticsReporter AnalyticsReporter, cfg config.Config) *SourceExecutor {
+	return &SourceExecutor{
+		log:               log,
+		analyticsReporter: analyticsReporter,
+		cfg:               cfg,
+	}
+}
+
+// Commands returns slice of commands the executor supports
+func (e *SourceExecutor) Commands() map[CommandVerb]CommandFn {
+	return map[CommandVerb]CommandFn{
+		CommandList: e.List,
+	}
+}
+
+// FeatureName returns the name and aliases of the feature provided by this executor
+func (e *SourceExecutor) FeatureName() FeatureName {
+	return sourceFeatureName
+}
+
+// List returns a tabular representation of Executors
+func (e *SourceExecutor) List(ctx context.Context, cmdCtx CommandContext) (interactive.Message, error) {
+	cmdVerb, cmdRes := parseCmdVerb(cmdCtx.Args)
+	defer e.reportCommand(cmdVerb, cmdRes, cmdCtx.Conversation.CommandOrigin, cmdCtx.Platform)
+	e.log.Debug("List sources")
+	return respond(e.TabularOutput(cmdCtx.Conversation.SourceBindings), cmdCtx), nil
+}
+
+// TabularOutput sorts source groups by key and returns a printable table
+func (e *SourceExecutor) TabularOutput(bindings []string) string {
+	buf := new(bytes.Buffer)
+	w := tabwriter.NewWriter(buf, 5, 0, 1, ' ', 0)
+	fmt.Fprintln(w, "SOURCE\tENABLED\tDISPLAY NAME")
+	sort.Strings(bindings)
+	for _, k := range bindings {
+		s := e.cfg.Sources[k]
+		if len(s.Plugins) > 0 {
```

Comment:
This is mutually exclusive. Is there a case where under one group plugins and non-plugin sources would be defined?

Reply:
@mszostok has better context, but old notation should be gradually deprecated, and old notation have already their configurations in different level.